core/impl/src/main/java/com/blazebit/persistence/impl/CTEManager.java

```diff
@@ -108,12 +108,12 @@ void buildClause(StringBuilder sb) {
             sb.append(cte.name);
             sb.append('(');
 
-            final List<String> attributes = cte.attributes;
-            sb.append(attributes.get(0));
+            final List<String> columnNames = cte.columnNames;
```

Comment:
This change is what my question is about.

Reply:
Looks good. It makes sense that this is required for multi-column associations.

Reply:
I've actually changed this behaviour back, because I foreseen some issues with multi column types (ie. java money). Didn't get a chance to test it, but the new approach uses more of the existing stuff, and didn't break existing tests.